src/lightning/fabric/strategies/ddp.py

```diff
@@ -124,7 +124,13 @@ def setup_module(self, module: Module) -> DistributedDataParallel:
         """Wraps the model into a :class:`~torch.nn.parallel.distributed.DistributedDataParallel` module."""
         device_ids = self._determine_ddp_device_ids()
         # https://pytorch.org/docs/stable/notes/cuda.html#id5
-        ctx = torch.cuda.stream(torch.cuda.Stream()) if device_ids is not None else nullcontext()
+        ctx = (
+            getattr(torch, f"{self.root_device.type.split(':')[0]}").stream(
```

Comment:
I'd avoid doing this inline. Better to `getattr` and assign to a variable, and then use the variable.

My main question here is what is the contract that ensures that an accelerator has a concept of streams. Unless I read through the code, as a developer I wouldn't know that I need to register streams as `torch.mygpu.stream` and `torch.mygpu.Stream()`.

So we should either guard the strategy to only apply to `"cuda"`, or introduce a stream contract to the accelerator. I'd much rather do the former.

Reply:
Since I see that accelerators such as Intel and MUSA (some GPGPU hardware) support streams, I think it is possible to provide such support for multiple backends. If torch.mygpu.Stream() is not supported, it can fall back to a non-stream implementation. I will improve this in my next commit.